scripts/release/homebrew/upload.sh

```diff
@@ -11,7 +11,7 @@ CLI_VERSION=`cat $SYSTEM_ARTIFACTSDIRECTORY/metadata/version`
 TAR_NAME=azure-cli-$CLI_VERSION.tar.gz
 
 mkdir -p $BUILD_STAGINGDIRECTORY/github/ >/dev/null 2>&1
-curl -sL https://github.com/Azure/azure-cli/archive/$TAR_NAME -o $BUILD_STAGINGDIRECTORY/github/$TAR_NAME
+curl -sL https://github.com/Azure/azure-cli/archive/refs/tags/$TAR_NAME -o $BUILD_STAGINGDIRECTORY/github/$TAR_NAME
```

Comment:
I think this file is deprecated. Could you please revert this change? @jiasli for confirmation.

---------------

In https://github.com/Azure/azure-cli/pull/8178, we uses GitHub url as `HOMEBREW_UPSTREAM_URL`, so `scripts/release/homebrew/upload.sh` is useless.

We also move the formula generation logic to https://github.com/Azure/azure-cli/blob/22feb5a5dcb1e23f0c1f1a444ff99af5c5502b19/azure-pipelines.yml#L597-L610
, so `scripts/release/homebrew/pipeline.sh` can also be removed.